src/shims/foreign_items.rs

```diff
@@ -671,6 +671,21 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
 
             // Architecture-specific shims
+            "llvm.x86.addcarry.64" if this.tcx.sess.target.arch == "x86_64" => {
```

Comment:
Is this also supported on 32 bit targets?

Reply:
I would guess not as this is a 64bit operation.

Reply:
It's not, as far as I could tell. I had it implemented on both 32 bit and 64 bit originally (ac99287a748a2b0469d4136fc88c5483421f629f) but the CI failed.

https://github.com/rust-lang/miri/runs/6235730002?check_suite_focus=true

```console
error[E0425]: cannot find function `_addcarry_u64` in module `arch`
  --> $DIR/intrinsics-x86.rs:13:32
   |
13 |     let c_out = unsafe { arch::_addcarry_u64(c_in, a, b, &mut sum) };
   |                                ^^^^^^^^^^^^^ help: a function with a similar name exists: `_addcarry_u32`
   |
  ::: /home/runner/.rustup/toolchains/master/lib/rustlib/src/rust/library/core/src/../../stdarch/crates/core_arch/src/x86/adx.rs:20:1
   |
20 | pub unsafe fn _addcarry_u32(c_in: u8, a: u32, b: u32, out: &mut u32) -> u8 {
   | -------------------------------------------------------------------------- similarly named function `_addcarry_u32` defined here
```

Now perhaps `llvm.x86.addcarry.64` exists and it's only the `_addcarry_u64` wrapper in stdarch which does not exist, but then we're without any good way to write tests for it.

Reply:
Right. We'll let libstd decide whether this should exist and make miri support it once/if that happens